docs/reference/api/resources/sources/list_source.rst

```diff
@@ -74,8 +76,8 @@ used as a data source. This means they must provide:
 
 * ``__getitem__(self, index)`` returning the item at position ``index`` of the list.
 
-A custom ListSource must also generate ``insert``, ``remove`` and ``clear``
-notifications when items are added or removed from the source.
+A custom ListSource must also inherit from :any:`Source`, and generate ``insert``,
```

Comment:
This is an interesting case of a subtle duck-typing change. The previous implementation would have allowed any object that honoured the (implicit) Source protocol; as a result of the tweak you made to core, the default behavior is "list like", and a Source must *literally* inherit from source, and any other type will be coerced (to varying degrees of success) into a ListSource. This then implies that ListSource will be robust to all possible input types, which I don't think is necessarily true.

Reply:
After discussion, we agreed:
* The ListSource constructor already accepts any iterable.
* We need some way for the user to distinguish an iterable which is to be used as a Source directly, versus copied into a new ListSource. Inheriting the Source class is the simplest way of doing this.
* The statement "Any object that adheres to the [collections.abc.MutableSequence](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableSequence) protocol can be used as a data source" isn't really true, because:
  * The `find` method isn't part of the standard Sequence API.
  * There are also differences in whether the other methods use identity versus equality.
  
  So this wording should instead refer to the ListSource API directly, and that API documentation should include all the relevant dunder methods. Later we may redefine it as a Protocol.